doc/trajopt_planner/trajopt_planner_tutorial.rst

```diff
@@ -0,0 +1,119 @@
+
+- convex approximation of *f(x)* and *g(x)* over the trust region. Then the approximated ones are converted to penalty functions.
+- affine approximation of *h(x)* over the trust region. Then the approximated *h(x)* is convetered to penalty function by considering its absolute value.
+
+ The following figure shows how TrajOpt algorithm works
+
+.. image:: algorithm.png
+   :width: 400px
+
+The bold italic parameters are the ones loaded from yaml file. All the constraints including obstacle avoidance, joint limits and target poses in joint space or Cartesian space are conveted to penalty functions so they make the final objective function. The most inner loop is where sequential quadratic programming is used as a trust region method. It calculates the second-order Tyler approximation in a box trust region.
+
+One important part of TrajOpt is how the obstacle avoidance constraint is formulated. In discrete case, the constraint is basically the difference between the signed distance (between robot link with itself or with obstacles) and a safe value and in the continuous case, the signed distance is between convex hull of two waypoints and obstacles.
+
+**Note:** The current implementation of trajopt in MoveIt supports constraints in joint space only.
+
+Getting Started
+---------------
+If you haven't already done so, make sure you've completed the steps in `Getting Started <../getting_started/getting_started.html>`_.
+
+You should also have gone through the steps in `Visualization with MoveIt RViz Plugin <../quickstart_in_rviz/quickstart_in_rviz_tutorial.html>`_
+
+Prerequisites
+--------------
+To use TrajOpt with your robot you must already have a MoveIt configuration package for your robot. For example, if you have a Panda robot, it's called ``panda_moveit_config``. This is typically configured using the `MoveIt Setup Assistant <../setup_assistant/setup_assistant_tutorial.html>`_.
+
+**Note:** in this tutorial, we use ``panda_moveit_config`` from the `ros-planning/panda_moveit_config <https://github.com/ros-planning/panda_moveit_config>`_ repository which contains the necessary config files for trajopt planner.
+
+Running the Demo
+----------------
+If you have the ``panda_moveit_config`` from the `ros-planning/panda_moveit_config <https://github.com/ros-planning/panda_moveit_config>`_ repository you should be able to simply run the example: ::
+
+  roslaunch moveit_tutorials trajopt_example_launch.launch
+
+This will run ``demo.launch`` from ``panda_moveit_config`` with passing ``trajopt`` for ``pipeline`` argument as the planner and runs the ``trajopt_example`` node.
+
+How TrajOpt works
+-----------------
+Motion planning problem in TrajOpt is defined by a set of cost (COST) and constraints (CNT) functions that are added to ``TrajOptProblem`` through ``ConstructProblem`` function. This function gets the information regarding to the problem (``ProblemInfo``) which carries different types of information explained below:
```

Comment:
Don't abbrev cost and constraints

Reply:
I'm actually ok with the abbreviation, since it relates to the specific spelling of an enum in the code.

Reply:
In trajopt, we have term_type which can have values like TT_CNT and TT_COST. These abbreviations are used a lot in trajopt so I think it is good to keep them here as well to be consistent with the code.